packages/security/src/policy.ts

```
@@ -0,0 +1,105 @@
+
+function permissionGateRule(
+  checkPermission: (subject: string, action: string) => boolean,
+): PolicyRule {
+  return ({ subject, action }) => {
+    if (!action) return;
+    const ok = checkPermission(subject, action);
+    if (!ok) throw new NotAllowedError('Permission denied');
+  };
+}
+
+function providerAccessRule(rule: ProviderAccessRule): PolicyRule {
```

ESLint

[error] Parameter should have an immutability of at least "ReadonlyDeep" (actual: "Mutable").

(functional/prefer-immutable-types)